core/src/main/scala/kafka/network/SocketServer.scala

```diff
@@ -528,6 +567,8 @@ private[kafka] class Processor(val id: Int,
     override def toString: String = s"$localHost:$localPort-$remoteHost:$remotePort-$index"
   }
 
+  // the receivesProcessed counter is defined only for testing
+  private[network] var receivesProcessed: Long = 0L
```

Comment:
We generally don't want to define variable and state just for unit tests. Can you find a way to test the code without adding such variables?